enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -1186,12 +1188,21 @@ func (s *SchedulerServer) enqueueTaskReservations(ctx context.Context, enqueueRe
 				// (in subsequent loop iterations) if the preferred node probe fails.
 				preferredNode = nil
 			} else {
-				sampleConnectedExecutors := opts.alwaysScheduleLocally
-				newNodes, err := nodeBalancer.SampleNodes(ctx, probeCount, sampleConnectedExecutors)
-				if err != nil {
+				nodes = nodeBalancer.nodes
+				if opts.alwaysScheduleLocally {
+					nodes = nodeBalancer.connectedExecutors
+				}
+				if len(nodes) == 0 {
 					return status.UnavailableErrorf("No registered executors in pool %q with os %q with arch %q.", pool, os, arch)
 				}
-				nodes = newNodes
+				rankedNodes, err := s.taskRouter.RankNodes(ctx, cmd, remoteInstanceName, toNodeInterfaces(nodes))
+				if err != nil {
+					log.Errorf("Failed to rank nodes: %s", err)
```

Comment:
What do you think about changing the RankNodes interface to not return an error and instead having the task router log a message instead? Doesn't seem like there's anything the calling code can do with the error and it's a bit confusing to both return a valid list of nodes and an error.

Reply:
SG, done. I was originally thinking that it's nice for tests to be able to check the err, but it seems just as effective to only check the nodes that are returned.